parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -51,7 +51,7 @@ sealed trait ParquetField[T] extends Serializable {
   protected final def nonEmpty(v: T): Boolean = !isEmpty(v)
 
   def write(c: RecordConsumer, v: T)(cm: CaseMapper): Unit
-  def newConverter: TypeConverter[T]
+  def newConverter(writerSchema: Type): TypeConverter[T]
```

Comment:
Alternately, we could create an overloaded method like:

```scala
def newConverter(): TypeConverter = newConverter(false)
def newConverter(avroCompat: Boolean): TypeConverter = ...
```

Reply:
This will allow forward binary compatibility. Otherwise we should move this to the v0.8 base branch

Reply:
hmm. forward compat would be nice, but long-term I'd like to be able to phase out the whole AvroCompat option from Magnolify and just write grouped arrays by default... so I'd rather not encode it further into the Magnolify API (`def newConverter(avroCompat: Boolean)`). So I think we can keep this as is and I'll rebase onto v0.8 branch

Reply:
although now that I think about it more, the param here should be readSchema/requestedSchema, not `writerSchema`